gems/actionmailer/7.0/actionmailer.rbs

```diff
@@ -4,9 +4,9 @@ module ActionMailer
 
     def self.layout: (String) -> void
 
-    def mail: (**untyped) ?{ (untyped) -> void } -> MessageDelivery
+    def mail: (**untyped) ?{ (untyped) -> void } -> Mail::Message
 
-    def self.with: (**untyped) -> instance
+    def self.with: (**untyped) -> untyped
```

Comment:
The return type of `ActionMailer::Base.with` is `ActionMailer::Parameterized::Mailer`, not an instance of `ActionMailer::Base`.

But returning Parameterized::Mailer is not good way because it strongly depends on reflection techniques to define user defined mailer methods. Therefore it's difficult to provide correct types for the `.with` usage.

This changes the return type of the `.with` method to untyped to keep code valid.

It would be better to create a generator for ActionMailer to support correct types for `.with` method.   

refs: https://github.com/rails/rails/blob/v7.0.8/actionmailer/lib/action_mailer/parameterized.rb#L101